third_party/nvfuser/csrc/ir_nodes.cpp

```diff
@@ -2196,14 +2196,15 @@ TensorDomain::TensorDomain(
     : Val(passkey, ValType::TensorDomain, DataType::Null),
       root_domain_(std::move(root_domain)),
       contiguity_(
-          contiguity.empty() ? std::vector<bool>(root_domain_.size(), false)
-                             : std::move(contiguity)) {
+          contiguity.empty()
+              ? std::vector<bool>(noBroadcasts(root_domain_).size(), false)
+              : std::move(contiguity)) {
   TORCH_CHECK(
-      contiguity_.size() == getMaybeRFactorDomain().size(),
+      contiguity_.size() == noBroadcasts(getMaybeRFactorDomain()).size(),
```

Comment:
Looks like we are expecting TensorDomain to receive a contiguity vector with adjusted length.

I'm not sure how this works without updating the TensorViewconstructor, where we have contiguity the length of torchscript tensor_type: https://github.com/csarofeen/pytorch/blob/28990ac6c23e931f2406626f3cd9ac56493d10f9/third_party/nvfuser/csrc/tensor_view.cpp#L128

Reply:
I think this function is updated